oak_dice/src/evidence.rs

```diff
@@ -74,8 +83,35 @@ pub struct RootLayerEvidence {
 }
 
 impl RootLayerEvidence {
-    pub fn get_tee_platform(&self) -> Option<TeePlatform> {
-        TeePlatform::from_repr(self.tee_platform)
+    pub fn get_tee_platform(&self) -> Result<TeePlatform, String> {
+        TeePlatform::from_repr(self.tee_platform).ok_or("invalid TEE Platform value".to_string())
+    }
+
+    pub fn get_remote_attestation_report(&self) -> Result<&[u8], String> {
+        match self.get_tee_platform()? {
+            TeePlatform::AmdSevSnp => {
+                if self.remote_attestation_report.len() < AMD_SEV_SNP_ATTESTATION_REPORT_SIZE {
```

Comment:
Is it ok to be larger than that value?

Reply:
I don't think this runtime check is needed. `remote_attestation_report` is an array with a fixed length that is longer than the AMD SEV-SNP attestation report in case the TDX report is longer. If we are worried that we might accidentally change the size of the array to be smaller than the attestation report size it would be better to check this at compile time using `static_assert`.

Reply:
The reasoning for the conditional was that this code would panic if it were larger. 

But @conradgrobler made a good point that this should be done statically. Updating...